vector/src/main/java/fr/gouv/tchap/activity/TchapRoomCreationActivity.java

```diff
@@ -166,13 +168,35 @@ private void hideKeyboard() {
         imm.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0);
     }
 
-    @OnClick(R.id.rly_hexagon_avatar)
-    void addRoomAvatar() {
+    private void openMediasPicker() {
         Intent intent = new Intent(TchapRoomCreationActivity.this, VectorMediasPickerActivity.class);
         intent.putExtra(VectorMediasPickerActivity.EXTRA_AVATAR_MODE, true);
         startActivityForResult(intent, REQ_CODE_UPDATE_ROOM_AVATAR);
     }
 
+    @OnClick(R.id.rly_hexagon_avatar)
+    void addRoomAvatar() {
+        if (!CommonActivityUtils.checkPermissions(CommonActivityUtils.PERMISSION_CAMERA, this)) {
+            
+        } else {
+            openMediasPicker();
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+
+        if (requestCode == CommonActivityUtils.PERMISSION_CAMERA) {
+            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                Log.d(LOG_TAG, "## camera permission granted");
+                openMediasPicker();
+            } else {
+                Log.d(LOG_TAG, "## camera permission denied");
+                openMediasPicker();
+            }
```

Comment:
Why don't we log "grantResults[0]" and call "openMediasPicker();" without if else?

Reply:
The log allow to know if the user has accepted or denied the camera access.
The openMediaPicker() method is called even if access to the camera has been denied by the user because it opens the mediaPicker activity. This screen allows the user to take a photo or choose a photo in his gallery. So even if access to the camera is denied, the user must still be able to upload a photo from his gallery.

I reversed the condition and moved the openMediasPicker()outside the if / else block.